### PR TITLE
guest_agent: Add function of guest-fsfreeze-freeze-list

### PR DESCRIPTION
**Pull request by xiagao**
Take guest-fsfreeze-freeze-list and guest-fsfreeze-freeze to one
function.
ID: 1817348
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
@lolyu  Call for your review..

File: virttest/guest_agent.py [hunk 1]
```diff
@@ -678,22 +678,35 @@ def verify_fsfreeze_status(self, expected):
...
+        Freeze File system on guest, there are two commands,
+        "guest-fsfreeze-freeze" and "guest-fsfreeze-freeze-list",
+         the second one can freeze some of mountpoints.
 
         :param check_status: Force this function to check the fsfreeze status
                              before/after sending cmd.
+        :param fsfreeze_list: Freeze fs with guest-fsfreeze-freeze or
+                              guest-fsfreeze-freeze-list
+        :param mountpoints: an array of mountpoints of filesystems to be frozen
+                            If omitted, every mounted filesystem is frozen
```
Comment (vivianQizhu) [inline virttest/guest_agent.py:693]: Please take care of punctuation and case.

File: virttest/guest_agent.py [hunk 2]
```diff
@@ -678,22 +678,35 @@ def verify_fsfreeze_status(self, expected):
...
 
     @error_context.context_aware
-    def fsfreeze(self, check_status=True, timeout=FSFREEZE_TIMEOUT):
+    def fsfreeze(self, check_status=True, timeout=FSFREEZE_TIMEOUT,
+                 fsfreeze_list=False, mountpoints=None):
         """
-        Freeze File system on guest.
+        Freeze File system on guest, there are two commands,
+        "guest-fsfreeze-freeze" and "guest-fsfreeze-freeze-list",
+         the second one can freeze some of mountpoints.
```
Comment (vivianQizhu) [inline virttest/guest_agent.py:686]: 'the second one allow to freeze by mount points'?

Comment (xiagao) [reply virttest/guest_agent.py]: Updated.

File: virttest/guest_agent.py [hunk 3]
```diff
@@ -678,22 +678,38 @@ def verify_fsfreeze_status(self, expected):
...
+        "guest-fsfreeze-freeze" and "guest-fsfreeze-freeze-list".
+        guest-fsfreeze-freeze: Sync and freeze all freezable,
+        local guest filesystems.
+        guest-fsfreeze-freeze-list: Sync and freeze specified guest
+        filesystems
 
         :param check_status: Force this function to check the fsfreeze status
                              before/after sending cmd.
+        :param fsfreeze_list: Freeze fs with guest-fsfreeze-freeze or
+                              guest-fsfreeze-freeze-list.
```
Comment (vivianQizhu) [inline virttest/guest_agent.py:694]: `fsfreeze_list` should apply to `guest-fsfreeze-freeze-list` only, not  `guest-fsfreeze-freeze`, right?

Comment (xiagao) [reply virttest/guest_agent.py:694]: yes, if fsfreeze_list is False, then use guest-fsfreeze-freeze command, else use guest-fsfreeze-freeze-list command.

Comment (vivianQizhu) [reply virttest/guest_agent.py:694]: @xiagao I mean the description might confuse users: `Freeze fs with guest-fsfreeze-freeze or guest-fsfreeze-freeze-list.`

Comment (xiagao) [reply virttest/guest_agent.py]: I update just now ,could you check if it's better?

**Pull request merged**